src/vl53_main.c

```diff
@@ -0,0 +1,288 @@
+    status = VL53L1X_SensorInit(VL53_I2C_address);
+    status += VL53L1X_SetDistanceMode(VL53_I2C_address, LONG_RANGE);
+    status += VL53L1X_SetTimingBudgetInMs(VL53_I2C_address, SPEED_50_HZ);
+    status += VL53L1X_SetInterMeasurementInMs(VL53_I2C_address, SPEED_50_HZ);
+    status += VL53L1X_SetROI(VL53_I2C_address, 8, 16);
+
+// Print process, debug purposes
+#ifdef DEBUG
+    if (status == 0)
+        UART_PutStr((uint8_t *)"Setup complete\r\n");
+#endif
+
+    // Of any error occured, terminate process
+    if (status != 0)
+    {   
+        ERROR_send_warning("Error in VL53 Setup function\r\n");
+        return;
+    }
+}
+
+// Starts measuring each zone to put in algorithm
+// Wakes up sensor, and starts ranging (in while loop), injecting data in algorithm
+void VL53_start_measuring()
+{
+    static int center[2] = SPAD_CENTERS; // These are the spad center of the 2 8x16 zones
+    static int zone = 0;                 // Index for center array
+    static uint8_t data_ready = 0;
+
+#ifdef DEBUG
+    static uint8_t start_done = 0;
+    if (!start_done)
+        UART_PutStr((uint8_t *)"Start Ranging\r\n");
+    start_done = 1;
+#endif
+
+    // Wake up sensor
+    status = VL53L1X_StartRanging(VL53_I2C_address);
+
+    // While loop is needed, continuous 
+    while (vl53_enable)
+    {
+        data_ready = 0; // Set data_ready to 0 before reading
+        while (!data_ready)
+        {
+            status = VL53L1X_CheckForDataReady(VL53_I2C_address, &data_ready);
+        }
+        status += VL53L1X_GetRangeStatus(VL53_I2C_address, &range_status);
+        status += VL53L1X_GetDistance(VL53_I2C_address, &distance);
+        status += VL53L1X_ClearInterrupt(VL53_I2C_address); // Clear interrupt before next measurement is called
+
+        // If any error occured, terminate process
+        if (status)
+        {   
+            ERROR_send_warning("Error: could not read measurement\r\n");
+            return (-1);
+        }
+
+        // Wait a few milliseconds to make sure measurung is done, then change the ROI
+        HAL_Delay(5);
+        status = VL53L1X_SetROICenter(VL53_I2C_address, center[zone]);
+
+        // If any error occured, terminate process
+        if (status)
+        {
+            ERROR_send_warning("Error: could not set new zone\r\n");
+            return (-1);
+        }
+
+        // Inject the measured distance in the people counting algorithm, return new number of people
+        people_count = VL53_counting_algorithm(distance, zone);
+        // Check people_count for faulty value (i.e. when the value becomes negative)
+        people_count = VL53_check_people_count(people_count);
+        VL53_display_people_counter(people_count);
+
+        // Toggle zone integer between 0 and 1
+        zone++;
+        zone = zone % 2;
+    }
+}
+
+// Function calculates the amount of people in a room, needs to be called continuously to work
+int16_t VL53_counting_algorithm(uint16_t distance, uint8_t zone)

```

Comment:
dit is ook echt een monster van een functie. Is het niet mogelijk om deze iets op te splitsen?

Reply:
vant weekend